src/linux/main.c

```diff
@@ -81,8 +87,28 @@ main(int argc, char **argv)
         case 'I':
             serial = optarg;
             break;
+        case 'g':
+            pty_group_name = optarg;
+            break;
+        case 'm':
+            errno = 0; // see man 2 strtoul
+            unsigned long int input_pty_mode = strtoul(optarg, NULL, 8);
+            if (errno) {
+                fprintf(stderr, "unable to parse pty mode: %s", optarg);
+                return -1;
+            }
```

Comment:
FWIW, despite the man page recommendation I'm not sure it is a great idea to inspect `errno` without first checking the return value.  FWIW, since `strtoul()` will return `ULONG_MAX` and that will already report a "mode invalid" in your code, I'm not sure the `errno` checks are necessary.

Reply:
Looking at the man page again, it appears that the ULONG_MAX and ULONG_MIN clamping behavior is only for out of range failures
```
       The  strtoul() function returns either the result of the conversion or, if there was a leading minus sign, the
       negation of the result of the conversion represented as an unsigned value, unless  the  original  (nonnegated)
       value would overflow; in the latter case, strtoul() returns ULONG_MAX and sets errno to ERANGE.  Precisely the
       same holds for strtoull() (with ULLONG_MAX instead of ULONG_MAX).
```
but failed conversions return 0, and errno is the only indication that something failed.
```
       The implementation may also set errno to EINVAL in case no conversion was performed (no digits seen, and 0 re-
       turned).
```
Though it does say that
```
       Since  strtoul()  can legitimately return 0 or ULONG_MAX (ULLONG_MAX for strtoull()) on both success and fail-
       ure, the calling program should set errno to 0 before the call, and then determine if  an  error  occurred  by
       checking whether errno has a nonzero value after the call.
```

Reply:
Though, on second thought, I probably should stick that through errstr, or at least give the error

Reply:
> Looking at the man page again, it appears that the ULONG_MAX and ULONG_MIN clamping behavior is only for out of range failures

Oops.  My mistake.

-Kevin

Reply:
NP. I won't admit how many times I had to read that manpage the first time round.